### PR TITLE
Changed Graph structure + created default Graph Built

### PR DESCRIPTION
**Pull request by monkin77**
-  From what I saw we should get away from using templates for every member of the Graph, since that would force us to declare everything on header files (Problems with forward declarations). The only solution was to define the functions on the .cpp files indicating the specific class of the template, but that defies the whole purpose of templates, so might aswell get rid of it.

- Furthermore, I also created static attributes nextID so that we automatically have the IDs for the Vertices + Edges.
- Finnaly, created a default Graph Builder

### CODE REVIEW

File: src/utils/GraphBuidler.cpp [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+using namespace std;
+
+void defaultGraphBuilder(Graph &G){
+    Position *p1 = new Position(1, 1);
+    Position *p2 = new Position(2, 2);
+    Position *p3 = new Position(3, 3);
+
+    G.addVertex(*p1);
+    G.addVertex(*p2);
+    G.addVertex(*p3);
```
Comment (BrunoRosendo) [inline src/utils/GraphBuidler.cpp]: We don't need to to allocate memory dynamically here. We can should do this and not worry about freeing it later
```suggestion
void defaultGraphBuilder(Graph &G){
    G.addVertex(Position(1, 1));
    G.addVertex(Position(2, 2));
    G.addVertex(Position(3, 3));
```
`Java messing with our minds 🙃`

**Review by BrunoRosendo — CHANGES_REQUESTED** [review]
Everything looks good. I'm just suggesting a small change. BTW, did you correct the changes you requested in my last PR?

File: src/utils/GraphBuidler.cpp (hunk 1 above)
Comment (monkin77) [reply src/utils/GraphBuidler.cpp]: Oh forgot to change that. It was actually me trying to solve the problem with templates that sucked my life force. I'll fix it

**Comment by monkin77** [comment]
> Everything looks good. I'm just suggesting a small change. BTW, did you correct the changes you requested in my last PR?

Yep I did

**Review by BrunoRosendo — APPROVED** [review]

**Pull request merged**